packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -198,6 +198,17 @@ mixin SchedulerBinding on BindingBase, ServicesBinding {
     window.onDrawFrame = _handleDrawFrame;
     SystemChannels.lifecycle.setMessageHandler(_handleLifecycleMessage);
     readInitialLifecycleStateFromNativeWindow();
+
+    if (!kReleaseMode) {
+      int frameNumber = 0;
+
+      window.onReportTimings = (List<FrameTiming> timings) {
```

Comment:
If the app has already set a `window.onReportTimings` listener, will this make it not work? Maybe we should also call the original `onReportTimings` if it's not null.

Reply:
This is called pretty early in app initialization; I don't know how likely it is that we'll lose the race condition with user code.

If a user of the API is responsible for delegating to the last person who set `onReportTimings`, how would that user know to stop calling the previous closure when that closure was no longer interested in events?

If we're concerned about this, I think an option here is to adjust the API to use a broadcast stream based approach (`Stream<FrameTiming> get onFrameTimings`). Clients could add themselves as a listener to the stream, and remove themselves, without having to worry about other potential users of the API interfering with them.

Reply:
Right, I think I'll keep the current API for a simple engine-to-framework communication, and add a broadcast stream for flexibility later as a pure framework pull request.

Reply:
OK, makes sense.